tube/spark/translator.py

```diff
@@ -161,8 +163,44 @@ def get_direct_children(self, root_df):
             root_df = root_df.leftOuterJoin(child_df).mapValues(lambda x: merge_and_fill_empty_fields(x, n.fields))
         return root_df
 
+    def get_types(self):
+        mapping = self.parser.mapping
+        models = self.parser.models
+        root = self.parser.root
+
+        types = {}
+
+        for k, v in mapping[root].items():
+            if k == "_aggregated_props":
+                types.update({i["name"]: (float,) for i in v})
+
+            if k == "_flatten_props":
+                for i in v:
+                    a = get_properties_types(models, get_attribute_from_path(models, root, i['path']))
+                    for j in i['_props']:
+                        types[j] = a[j]
+
+            if k == "_props":
+                types.update({w: get_properties_types(models, root)[w] for w in v})
+
+        for k, v in types.items():
```

Comment:
can we move this part to another function called `choosing_type` or something like that

Reply:
Will do.